hasjob/models/jobpost.py

```diff
@@ -225,66 +229,49 @@ def expiry_date(self):
     def after_expiry_date(self):
         return self.expiry_date + timedelta(days=1)
 
-    def status_label(self):
-        if self.status == POSTSTATUS.DRAFT:
-            return _("Draft")
-        elif self.status == POSTSTATUS.PENDING:
-            return _("Pending")
-        elif self.is_new():
-            return _("New!")
-
-    def is_draft(self):
-        return self.status == POSTSTATUS.DRAFT
-
-    def is_pending(self):
-        return self.status == POSTSTATUS.PENDING
-
-    def is_unpublished(self):
-        return self.status in POSTSTATUS.UNPUBLISHED
-
-    def is_listed(self):
-        now = datetime.utcnow()
-        return (self.status in POSTSTATUS.LISTED) and (
-            self.datetime > now - agelimit)
-
-    def is_public(self):
-        return self.status in POSTSTATUS.LISTED
-
-    def is_flagged(self):
-        return self.status == POSTSTATUS.FLAGGED
-
-    def is_moderated(self):
-        return self.status == POSTSTATUS.MODERATED
-
-    def is_announcement(self):
-        return self.status == POSTSTATUS.ANNOUNCEMENT
-
-    def is_new(self):
-        return self.datetime >= datetime.utcnow() - newlimit
-
-    def is_closed(self):
-        return self.status == POSTSTATUS.CLOSED
-
-    def is_unacceptable(self):
-        return self.status in POSTSTATUS.UNACCEPTABLE
-
-    def is_old(self):
-        return self.datetime <= datetime.utcnow() - agelimit
-
-    def pay_type_label(self):
-        return PAY_TYPE.get(self.pay_type)
+    state.add_conditional_state('LISTED', state.PUBLIC, lambda jobpost: jobpost.datetime >= datetime.utcnow() - agelimit, label=('listed', __("Listed")))
+    state.add_conditional_state('CONFIRMABLE', state.UNPUBLISHED, lambda jobpost: jobpost.admin_is(current_auth.user), label=('confirmable', __("Confirmable")))
```

Comment:
This one is making me uncomfortable for the `current_auth` import. Ideally, the `admin_is` method should be replaced with an `edit` permission granted in `self.permissions`. This lambda can then become `lambda jobpost: jobpost.current_permissions.edit`